### PR TITLE
chore: cleanup of deno lint/ vscode "problems", etc

### PR DESCRIPTION
**Pull request by cscheid**
This adds a more robust deno.jsonc with a list of ignore targets, so that project-wide "deno lint" is fruitful.

This also adds `./package/src/common/project-lint.sh` which reports all files in the project that need linting.

### CODE REVIEW

File: src/command/render/render-files.ts [hunk 1]
```diff
@@ -288,8 +286,9 @@ export async function renderFiles(
       const file = files[i];
 
       if (progress) {
+        console.log("HERE");
```
Comment (dragonstyle) [inline src/command/render/render-files.ts:289]: 🤣

Comment (cscheid) [reply src/command/render/render-files.ts:289]: amazing.

File: dev-docs/update-deno_jsonc.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## tl;dr
+
+deno.jsonc currently doesn't support globs, so we have a script to expand the globs ourselves.
+
+## Steps
+
+```
+$ cd quarto-cli
+$ deno run --allow-all package/src/common/create-deno-config.ts > deno.jsonc
```
Comment (cderv) [inline dev-docs/update-deno_jsonc.md:9]: @cscheid is the ts file is missing ? I don't see it in the PR and not in my local clone of main

File: deno.jsonc [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+      ],
+      "exclude": [
+        "src/resources/create/extensions/revealjs-plugin/_extensions/qstart-filesafename-qend/qstart-filesafename-qend.js",
+        "src/resources/editor/",
+        "src/core/lib/external/",
+        "src/vendor/",
+        "src/resources/vendor/",
+        "src/resources/formats/pdf/pdfjs/",
+        "src/resources/deno_std/",
+        "/Users/cscheid/repos/github/quarto-dev/quarto-cli/package/pkg-working/share/extensions/quarto/video/resources/videojs/video.min.js",
```
Comment (cderv) [inline deno.jsonc:17]: @cscheid Is the absolute really paths expected ?

**Review by cderv — COMMENTED** [review]
Found this PR after trying to understand how to add a folder to be ignored in linting

How am I suppose to add to deno config ? seems like a file is missing and that files are no what I would expect ... 🤔

File: deno.jsonc (hunk 3 above)
Comment (cscheid) [reply deno.jsonc:17]: No, they should 100% be relative.

Comment (cderv) [reply deno.jsonc:17]: Fixed in 57760013b880406ea1821d3428e5e2c70c236011 

thanks

**Pull request merged**